packages/swing-store/src/transcriptStore.js

```diff
@@ -0,0 +1,436 @@
+    const sql = includeHistorical
+      ? sqlGetAllSpanMetadata
+      : sqlGetCurrentSpanMetadata;
+    for (const rec of sql.iterate()) {
+      yield [historicSpanMetadataKey(rec), JSON.stringify(rec)];
+    }
+  }
+
+  /**
+   * Obtain artifact names for spans contained in this store.
+   *
+   * @param {boolean} includeHistorical If true, include all spans that are
+   * present in the store regardless of their currency; if false, only include
+   * the current span for each vat.
+   *
+   * @yields {string}
+   * @returns {Iterable<string>}  An iterator over the names of all the artifacts requested
+   */
+  function* getArtifactNames(includeHistorical) {
+    const sql = includeHistorical
+      ? sqlGetAllSpanMetadata
+      : sqlGetCurrentSpanMetadata;
+    for (const rec of sql.iterate()) {
+      yield spanArtifactName(rec);
+    }
+  }
+
+  const sqlGetSpanEndPos = db.prepare(`
+    SELECT endPos
+    FROM transcriptSpans
+    WHERE vatID = ? AND startPos = ?
+  `);
+  sqlGetSpanEndPos.pluck(true);
+
+  const sqlReadSpanItems = db.prepare(`
+    SELECT item
+    FROM transcriptItems
+    WHERE vatID = ? AND ? <= position AND position < ?
+    ORDER BY position
+  `);
+
+  /**
+   * Read the items in a transcript span
+   *
+   * @param {string} vatID  The vat whose transcript is being read
+   * @param {number} [startPos] A start position identifying the span to be
+   *    read; defaults to the current span, whatever it is
+   *
+   * @returns {Iterable<string>}  An iterator over the items in the indicated span
+   */
+  function readSpan(vatID, startPos) {
+    let endPos;
+    if (startPos === undefined) {
+      ({ startPos, endPos } = getCurrentSpanBounds(vatID));
+    } else {
+      assert.typeof(startPos, 'number');
+      endPos = sqlGetSpanEndPos.get(vatID, startPos);
+      assert.typeof(
+        endPos,
+        'number',
+        `no transcript span for ${vatID} at ${startPos}`,
+      );
+    }
+    insistTranscriptPosition(startPos);
+    startPos <= endPos || Fail`${q(startPos)} <= ${q(endPos)}}`;
+
+    function* reader() {
+      for (const { item } of sqlReadSpanItems.iterate(
+        vatID,
+        startPos,
+        endPos,
+      )) {
+        yield item;
+      }
+    }
+
+    if (startPos === endPos) {
+      return empty();
+    }
+
+    return reader();
+  }
+
+  const nl = Buffer.from('\n');
+
+  /**
+   * Read a transcript span and return it as a stream of data suitable for
+   * export to another store.  Items are terminated by newlines.
+   *
+   * @param {string} vatID  The vat whose transcript is being read
+   * @param {number} [startPos] A start position identifying the span to be read
+   *
+   * @returns {AsyncIterable<Uint8Array>}
+   * @yields {Uint8Array}
+   */
+  async function* exportSpan(vatID, startPos) {
+    for (const entry of readSpan(vatID, startPos)) {
+      const item = typeof entry === 'string' ? Buffer.from(entry) : entry;
+      yield item;
+      yield nl;
```

Comment:
I'm still confused how `entry` can ever be something else than a string. In which case I'd prefer ``yield Buffer.from(`${entry}\n`)`` given the weird node stream semantics of how this will likely be consumed.

Reply:
I think this is an artifact of the evolution path that this code took, as there actually was a time when sometimes it could actually be a `Buffer`, which is what caused the code you're questioning to originally be summoned into existence. I think this may have been when sometimes the span a whole was delivered as a compressed blob (which _is_ a buffer) and so the code that consumed it also had to traffic in Buffers.  The now irrelevant particulars are lost to the fog of memory.  Now, however, it's strictly iterating database records where the field in question is always a string, so the simplification you propose should work fine.